client/my-sites/stats/stats-post-summary/index.jsx

```diff
@@ -0,0 +1,163 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { localize } from 'i18n-calypso';
+import { connect } from 'react-redux';
+import { findIndex, findLastIndex, flatten, flowRight, get, range } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import SummaryChart from '../stats-summary';
+import SectionNav from 'components/section-nav';
+import NavTabs from 'components/section-nav/tabs';
+import NavItem from 'components/section-nav/item';
+import QueryPostStats from 'components/data/query-post-stats';
+import { getPostStats } from 'state/stats/posts/selectors';
+
+class StatsPostSummary extends Component {
+	static propTypes = {
+		postId: PropTypes.number,
+		siteId: PropTypes.number,
+		translate: PropTypes.func,
+	};
+
+	state = {
+		period: 'day',
+	};
+
+	selectPeriod( period ) {
+		return () => this.setState( {
+			selectedRecord: undefined,
+			period
+		} );
+	}
+
+	selectRecord = record => {
+		this.setState( { selectedRecord: record } );
+	};
+
+	getChartData() {
```

Comment:
I thought about extracting this method to a selector. The drawback is having to import `moment` directly in the selector. Thoughts ?, I could avoid it by keeping the moment calls here, but I don't think it's a good idea since we'll still have the`switch` here as well as in the selector. Thoughts?